.github/workflows/lock-dependency.yml

```diff
@@ -91,7 +91,7 @@ jobs:
     steps:
       - uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
         with:
-          persist-credentials: false
+          persist-credentials: true
```

Comment:
Pleas add a comment explaining why (and by what) it is needed

Reply:
https://github.com/DataDog/dd-trace-rb/pull/4488/commits/41d083e1484d881e0d4660ceec3f4789dbb21ce4